### PR TITLE
reenable legacy spellchecker due to blacklisted dictionaries

### LINKED ISSUE #7018: Problems with spell check Macintosh
Trying to use Spell Check under MacOs Catalina Rstudio Version 1.3.959

The following error pops up:

check spelling An error has occurred: (TypeError) : Cannot read property 'check' of undefined

The erros happens whichever language I choose for checking

### PR DESCRIPTION
**Pull request by adamconroy**
I had forgotten that I changed the manual spellchecker to also use Typo.js when I had to blacklist some dictionaries and that broke manual spellchecking for those dictionaries entirely. Bringing back the old CPP Hunspell until we can fix 6116.

Fixes #7018

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/common/spelling/TypoSpellChecker.java [hunk 1]
```diff
@@ -274,6 +275,15 @@ private boolean checkCustomDicts(String word)
 
    public void checkSpelling(List<String> words, final ServerRequestCallback<SpellCheckerResult> callback)
    {
+      // If realtime is turned off call the backend spellchecker because we haven't loaded a
+      // dictionary on the frontend. Ideally we DO load the frontend dictionary but currently
+      // Typo.js isn't compatible with some of our dictionaries (see: TypoSpellChecker.realtimeBlacklist)
+      // so we need to keep the legacy call for now.
+      if (!userPrefs_.realTimeSpellchecking().getValue()) {
```
Comment (jmcphers) [inline src/gwt/src/org/rstudio/studio/client/common/spelling/TypoSpellChecker.java:282]: Nit: brace style

File: src/gwt/src/org/rstudio/studio/client/common/spelling/TypoSpellChecker.java [hunk 2]
```diff
@@ -372,7 +382,7 @@ private void loadDictionary()
    {
       final String language = userPrefs_.spellingDictionaryLanguage().getValue();
 
-      if (typoLoaded_ && loadedDict_.equals(language))
+      if (!userPrefs_.realTimeSpellchecking().getValue() || typoLoaded_ && loadedDict_.equals(language))
```
Comment (jmcphers) [inline src/gwt/src/org/rstudio/studio/client/common/spelling/TypoSpellChecker.java:385]: Could you use parens to group this so it's clear that the intent is to group the first two tests? (presuming that's the intent -- this isn't clear to me at first glance)

Comment (adamconroy) [reply src/gwt/src/org/rstudio/studio/client/common/spelling/TypoSpellChecker.java:385]: It is not the intent. We want to exit out of this function if either realtime is turned off or the dictionary is loaded AND it's the correct dictionary. I'll put it on another line so it's a bit more clear. This is actually just reverting it to the exact same code from before the prior PR to here.

File: src/gwt/src/org/rstudio/studio/client/common/spelling/TypoSpellChecker.java (hunk 1 above)
Comment (adamconroy) [reply src/gwt/src/org/rstudio/studio/client/common/spelling/TypoSpellChecker.java:282]: You get too used to using your preferred brace style in JavaScript for a while and this happens...

**Pull request merged**